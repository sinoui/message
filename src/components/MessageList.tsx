import React from 'react';
import Slide from 'sinoui-components/transitions/Slide';
import { TransitionGroup } from 'react-transition-group';
import { MessageInterface } from '../types';
import MessageListWrapper from './MessageListWrapper';
import Message from './Message';
import MessageIcon from './MessageIcon';

function MessageList({ messages }: { messages: MessageInterface[] }) {
  return (
    <TransitionGroup component={MessageListWrapper}>
      {messages.map((message) => (
        <Slide key={message.key} appear timeout={300} in direction="down">
          <Message>
            <MessageIcon type={message.type} />
            <div>{message.content}</div>
          </Message>
        </Slide>
      ))}
    </TransitionGroup>
  );
}

export default MessageList;
