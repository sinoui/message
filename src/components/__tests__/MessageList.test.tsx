import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import TestWrapper from './TestWrapper';
import MessageList from '../MessageList';

describe('MessageList 镜像测试', () => {
  it('渲染MessageList--无数据', () => {
    const tree = renderer
      .create(
        <TestWrapper>
          <MessageList messages={[]} />
        </TestWrapper>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
