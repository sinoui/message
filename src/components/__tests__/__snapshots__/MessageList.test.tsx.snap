// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageList 镜像测试 渲染MessageList 1`] = `
.c0 {
  position: fixed;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0 8px;
  margin-top: -40px;
  top: 16px;
  right: 0;
  left: 0;
  z-index: 2900;
  line-height: 1.5;
  font-size: 0.875rem;
  font-family: "Helvetica Neue",Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  color: rgba(0,0,0,0.87);
  pointer-events: none;
  box-sizing: border-box;
}

<div
  className="c0"
/>
`;

exports[`MessageList 镜像测试 渲染MessageList--无数据 1`] = `
.c0 {
  position: fixed;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0 8px;
  margin-top: -40px;
  top: 16px;
  right: 0;
  left: 0;
  z-index: 2900;
  line-height: 1.5;
  font-size: 0.875rem;
  font-family: "Helvetica Neue",Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  color: rgba(0,0,0,0.87);
  pointer-events: none;
  box-sizing: border-box;
}

<div
  className="c0"
/>
`;
